impl/src/main/java/org/ehcache/impl/internal/store/disk/OffHeapDiskStore.java

```diff
@@ -488,18 +490,25 @@ public void initAuthoritativeTier(AuthoritativeTier<?, ?> resource) {
    */
   @SuppressWarnings("unchecked")
   public static <T> PersistentPortability<T> persistent(final Portability<T> normal) {
-    final Class<?> normalKlazz = normal.getClass();
-    Class<?>[] delegateInterfaces = normalKlazz.getInterfaces();
-    Class<?>[] proxyInterfaces = Arrays.copyOf(delegateInterfaces, delegateInterfaces.length + 1);
-    proxyInterfaces[delegateInterfaces.length] = PersistentPortability.class;
-
-    return (PersistentPortability<T>) Proxy.newProxyInstance(normal.getClass().getClassLoader(), proxyInterfaces, (o, method, os) -> {
-      if (method.getDeclaringClass().equals(Persistent.class)) {
-        return null;
-      } else {
-        return method.invoke(normal, os);
+    if (normal instanceof PersistentPortability<?>) {
+      return (PersistentPortability<T>) normal;
+    } else {
+      Set<Class<?>> proxyInterfaces = new HashSet<>();
```

Comment:
I'd still encourage you to use a `LinkedHashSet` here since using a normal `HashSet` is randomizing the interface order.  But I'm not familiar with the objects that would be passed to this method so if you know that classes passed in won't have method collisions, I can go with this.

Reply:
I would go with a `LinkedHashSet` for extra safety; at least bug in there would be consistent across JVM implementations.